pkg/storegateway/indexheader/header.go

```diff
@@ -46,9 +46,11 @@ type Reader interface {
 type Config struct {
 	MapPopulateEnabled  bool `yaml:"map_populate_enabled" category:"experimental"`
 	StreamReaderEnabled bool `yaml:"stream_reader_enabled" category:"experimental"`
+	FileHandlePoolSize  uint `yaml:"file_handle_pool_size" category:"experimental"`
 }
 
 func (cfg *Config) RegisterFlagsWithPrefix(f *flag.FlagSet, prefix string) {
 	f.BoolVar(&cfg.MapPopulateEnabled, prefix+"map-populate-enabled", false, "If enabled, the store-gateway will attempt to pre-populate the file system cache when memory-mapping index-header files.")
 	f.BoolVar(&cfg.StreamReaderEnabled, prefix+"stream-reader-enabled", false, "If enabled, the store-gateway will use an experimental streaming reader to load and parse index-header files.")
+	f.UintVar(&cfg.FileHandlePoolSize, prefix+"file-handle-pool-size", 1, "Maximum number of file handles the store-gateway keeps open for each index-header file when using the streaming reader.")
```

Comment:
[nit] A couple of suggestions. If you agree with the renaming, then the YAML field should be renamed too.

```suggestion
	f.UintVar(&cfg.FileHandlePoolSize, prefix+"stream-reader-max-idle-file-handles", 1, "Maximum number of idle file handles the store-gateway keeps open for each index-header file when using the streaming reader. This option is used only when the index-header streaming reader is enabled.")
```